pkg/cmd/search/repos/repos.go

```diff
@@ -0,0 +1,238 @@
+package repos
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/pkg/cmd/search/shared"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/export"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/cli/cli/v2/pkg/search"
+	"github.com/cli/cli/v2/pkg/text"
+	"github.com/cli/cli/v2/utils"
+	"github.com/spf13/cobra"
+)
+
+type ReposOptions struct {
+	Browser      cmdutil.Browser
+	Config       func() (config.Config, error)
+	Exporter     cmdutil.Exporter
+	GoTemplate   string
+	HttpClient   func() (*http.Client, error)
+	IO           *iostreams.IOStreams
+	JqExpression string
+	Query        search.Query
+	WebMode      bool
+}
+
+func NewCmdRepos(f *cmdutil.Factory, runF func(*ReposOptions) error) *cobra.Command {
+	opts := &ReposOptions{
+		Browser:    f.Browser,
+		Config:     f.Config,
+		HttpClient: f.HttpClient,
+		IO:         f.IOStreams,
+		Query:      NewSearchQuery(),
+	}
+
+	cmd := &cobra.Command{
+		Use:   "repos [<query>]",
+		Short: "Search repositories",
+		RunE: func(c *cobra.Command, args []string) error {
+			opts.Query.Keywords = args
+			err := cmdutil.MutuallyExclusive("expected exactly one of `--jq`, `--template`, or `--web`",
+				opts.GoTemplate != "",
+				opts.JqExpression != "",
+				opts.WebMode)
+			if err != nil {
+				return err
+			}
+			if opts.Query.Limit < 1 || opts.Query.Limit > 1000 {
+				return cmdutil.FlagErrorf("`--limit` must be between 1 and 1000")
+			}
+			if runF != nil {
+				return runF(opts)
+			}
+			return reposRun(opts)
+		},
+	}
+
+	// Output flags
+	cmd.Flags().StringVarP(&opts.GoTemplate, "template", "t", "", "Format JSON output using a Go template")
+	cmd.Flags().StringVarP(&opts.JqExpression, "jq", "q", "", "Format JSON output using a jq `expression`")
+	cmd.Flags().BoolVarP(&opts.WebMode, "web", "w", false, "Open the query in the web browser")
+
+	// Query parameter flags
+	cmd.Flags().IntVarP(&opts.Query.Limit, "limit", "L", 30, "Maximum number of repositories to fetch")
+	cmd.Flags().Var(opts.Query.Order, "order", "Order of repositories returned, ignored unless '--sort' is specified")
+	cmd.Flags().Var(opts.Query.Sort, "sort", "Sorts the repositories by stars, forks, help-wanted-issues, or updated")
+
+	// Query qualifier flags
```

Comment:
For context, I decided to use these custom qualifier flag types mainly because I realized that many of the qualifiers need validation, for example making sure that the value for the visibility qualifier is either `public`, or `private` and invalid in all other cases. This is not supported with the `pflag` library so I created my own. They support arbitrary validation functions so they are extendable to other use cases if we like this pattern.